### ISSUE TITLE
Test function name starts with `test_`

### ISSUE THREAD

**Issue by laralove143** [rust-lang/rust-clippy#8931]
### What it does

None of the examples in [Rust guidelines](https://doc.rust-lang.org/book/ch11-01-writing-tests.html) on writing tests start functions with `test_`, which makes for better code quality since `tests::foo` is more readable than `tests::test_foo`

Somewhat related to: [clippy::module_name_repetitions](https://rust-lang.github.io/rust-clippy/master/#module_name_repetitions)

### Lint Name

test_name_repetitions

### Category

pedantic

### Advantage

Make the code match Rust guidelines thus making it more idiomatic

### Drawbacks

Not sure

### Example

```rust
#[cfg(test)]
mod tests {
    #[test]
    fn test_foo() {
        assert!(true);
    }
}
```

Could be written as:

```rust
rust
#[cfg(test)]
mod tests {
    #[test]
    fn foo() {
        assert!(true);
    }
}
```

**Comment by mikerite** [comment]
I've seen that sometimes the test functions are directly in the module that gets tested and not a nested module. I guess the lint isn't applicable in that case.

**Comment by laralove143** [comment]
Could you elaborate? Maybe showing a module tree

**Comment by laralove143** [comment]
Is anyone looking into this?

**Comment by kraktus** [comment]
> Could you elaborate? Maybe showing a module tree

Something like that 
```rust
struct Foo;

// test `Foo` in the middle of the file 
#[cfg(test)]
#[test]
fn test_foo() { todo!() }

struct Bar;

// test `Bar` in the middle of the file 
#[cfg(test)]
#[test]
fn test_bar() { todo!() }

// ...

```

**Comment by laralove143** [comment]
Oh I didn't know you could do that, we could just limit this lint for items in a module containing `test` in its name

**Comment by andrewbanchich** [comment]
i personally don't see a point to the `test_` prefix even if it isn't in a module containing `test` in the name.

if the function is a test, we can see

```rust
#[cfg(test)]
#[test] // it says test right here. it's the same thing, just moved up a line.
fn foo_works() { todo!() }
```
so it's obvious in the code. and when we run tests, we see

```bash
test foo_works ... ok
```

so `cargo test` output already adds a `test` at the beginning. plus, you just ran `cargo test` so it should be obvious that the functions you're looking at are tests.

**Comment by Caylub** [comment]
@farazdagi Yeah, this is my bad, I opened this with good intentions, and never followed through.

Feel free to pick this up!
